### ISSUE TITLE
Finding differentially expressing genes in a subgroup of cells

### ISSUE THREAD

**Issue by kathirij** [satijalab/seurat#285]
Hello all,
  I am interested in determining differentially expressing genes in a small group of cells that does not cluster on its own. These cells are spread over 9 different clusters (the entire object has 13 clusters total). I would like to compare the transcriptome of this small group of cells with the remaining cells to see what makes them unique, if anything. Is there a way to feed "cells.use" or something similar in the FindMarkers command? Alternatively, could I add new ids to a group of cells in anyway so that FindMarkers could distinguish between the two types of cells?

Thanks

**Comment by leonfodoulian** [comment]
Hi,

You can specifically change the identity of these cells using the `SetIdent()` function of Seurat, and leave the identity of all other cells unchanged. This will allow you to use these new identities to compare your cells to each other.

```R
# Create a vector of cells of interest
cells.of.interest <- c("cell_names")

# Change the identity of the cells of interest to 'special_cells'
object <- SetIdent(object = object, cells.use = cells.of.interest, id = "special_cells")
```

Alternatively, you can change the identity of all your cells, by defining a new column in your `object@meta.data` slot in such a way that one level is attributed to the `cells.of.interest`, and another for all other cells. You can then change the identity of all the cells by using the `SetAllIdent()` function of Seurat.

```R
# Create a vector of cells of interest
cells.of.interest <- c("cell_names")

# Create a new column in the object@meta.data slot defining two groups of cells:
# those of interest, and all the others
object@meta.data$cell.type <- ifelse(rownames(object@meta.data) %in% cells.of.interest, "of_interest", "other")

# Change the identity of all the cells to attributes of object@meta.data$cell.type
object <- SetAllIdent(object = object, id = "cell.type")
```

The use of one or the other of the above mentioned approaches will depend on how you want to perform your analysis (i.e. whether it is by comparing these cells to multiple distinct cell types, or to all other cells irrespective of their differences). 

Hope this was clear!

Best,
Leon

**Comment by kathirij** [comment]
Thanks a lot!

the SetIdent function solved my issue. I had completely overlooked that function.

PS: you have solved so many of my issues related to Seurat that if we ever come across at a conference or something, beer is on me!

**Comment by leonfodoulian** [comment]
@kathirij 

Any reason why you reopened the issue?

> PS: you have solved so many of my issues related to Seurat that if we ever come across at a conference or something, beer is on me!

It's my pleasure. If we manage to find some good amber beer, then I'll be delighted!

**Comment by kathirij** [comment]
@leonfodoulian 
I meant to close the issue and must've reopened it by mistake.

Amber beer it is.

**Comment by haabbas** [comment]
Reopening this case as I third the above without full success. I have 4 single cell files, that I combined the 4 SeuratObjects using:

```
immune.anchors <- FindIntegrationAnchors(object.list = list(S1, S2, S3, S4), dims = 1:20)
immune.combined <- IntegrateData(anchorset = immune.anchors, dims = 1:20)
```

My CD8A cells are dispersed in different clusters. I would like to 1) Compare the CD8A in pair-wise fashion across different samples and 2) Attempt to define a new cluster using multiple markers i.e. "CD8A" high, "FOXP3" low ) etc.

For # 1 , I tried:

`cells.of.interest <- c("CD8A")
new_object <- SetIdent(object = immune.combined, cells.use = cells.of.interest, id = "special_cells", value="CytoT")
` 
 However, I was still unable to figure out how to do the comparisons for differential gene expression fo the CD8A cells between S1 and S2 or any other combination

For #2, I think I could use 
`WhichCells(object=immune.combined, expression= CD8A > 0) 
`
But the issues are 1) how to set the cut-off for expression? and 2) how to use multiple markers at same time.

Appreciate your thoughts on this
